OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -222,6 +221,24 @@ const OpenSim::PhysicalFrame& Joint::getParentFrame() const
     return getConnector<PhysicalFrame>("parent_frame").getConnectee();
 }
 
+Coordinate::MotionType Joint::getMotionType(CoordinateIndex cix) const
+{
+    OPENSIM_THROW_IF(cix >= _motionTypes.size(), Exception,
+        "Joint::getMotionType() given an invalid CoordinateIndex");
+    return _motionTypes[cix];
+}
+
+void Joint::setMotionType(CoordinateIndex cix, Coordinate::MotionType mt)
+{
+    OPENSIM_THROW_IF(cix >= numCoordinates(), Exception,
+        "Joint::setMotionType() for an invalid CoordinateIndex");
+    if (_motionTypes.size() <= cix)
```

Comment:
this method is primarily being used to set the `MotionType` as Coordinates are constructed during `Joint` construction which grows `_motionTypes` and will be ` < numCoordinates()` until it is added to the list.   `Joint::constructCoordinate()` checks (asserts) that the list of coordinates and motionTypes are are the same size after the motion type after this method is called.